src/TetMesh.cpp

```diff
@@ -119,3 +119,38 @@ bool wmtk::TetMesh::check_mesh_connectivity_validity() const
 
     return true;
 }
+
+bool wmtk::TetMesh::collapse_edge(const Tuple& t, std::vector<Tuple>& new_edges)
+{
+    // todo
+
+    return true;
+}
+
+
+std::vector<wmtk::TetMesh::Tuple> wmtk::TetMesh::get_vertices() const
+{
+    std::vector<TetMesh::Tuple> edges;
+    for (auto i = 0; i < m_vertex_connectivity.size(); i++) {
+        auto& vc = m_vertex_connectivity[i];
+        if (vc.m_is_removed) continue;
+        assert(!vc.m_conn_tets.empty());
+        auto tid = vc[0];
+        auto local_vid = m_tet_connectivity[tid].find(i);
+
+        // note: the following conversion of local_vid-eid is dependent on the specifics of
+        // m_local_edges
+        edges.emplace_back(tuple_from_edge(tid, local_vid));
+        if (local_vid == 3) edges.back() = switch_vertex(edges.back());
+        assert(edges.back().vid() == i);
```

Comment:
Should `Tuple::update_version_number(*this)` be called for every newly created tuple?  Otherwise, this tuple may be invalid from the birth...

Reply:
we should also add `assert(edges.back().is_version_number_valid(*this))`